cmd/config/set.go

```diff
@@ -86,6 +88,43 @@ func runSet(cmd *cobra.Command, vp *viper.Viper, key, value string) error {
 	if err != nil {
 		return fmt.Errorf("cannot assign value=%s for key=%s, expected type=%s: %w", value, key, typ, err)
 	}
-	vp.Set(key, newVal)
-	return vp.WriteConfig()
+
+	// Create a file only viper from the configured file to avoid writing
+	// defaults and/or values set via environment variables or flags.
+	// This viper config is only used to write the resulting config.
+	// This method also prevents from writing default values for all keys
+	// therefore only writing key/value pairs explicitely set by the caller.
+	configPath := vp.GetString("config")
+	fileVP, err := newFileOnlyViper(configPath)
+	if err != nil {
+		return err
+	}
+	fileVP.Set(key, newVal)
+	return fileVP.WriteConfigAs(configPath)
+}
+
+// newFileOnlyViper creates a new viper config that only reads from the given
+// configuration file and is not bound to any environment variable or flag.
+func newFileOnlyViper(configPath string) (*viper.Viper, error) {
+	if configPath == "" {
+		return nil, errors.New("config file undefined")
+	}
+
+	path := filepath.Clean(configPath)
+	base := filepath.Base(path)
+	ext := filepath.Ext(path)
+
+	dir := filepath.Dir(path)
+	filename := strings.TrimSuffix(base, ext)
+	typ := strings.TrimPrefix(ext, ".")
+
+	vp := viper.New()
+	vp.SetConfigName(filename)
+	vp.SetConfigType(typ)
+	vp.AddConfigPath(dir)
+
+	// tentatively read the config file
+	// it will fail if the config file does not exist, which is fine
+	_ = vp.ReadInConfig()
```

Comment:
I think there can be other errors (such as bad file format) in which case we do not silently want to ignore them, as it would cause use to accidentally overwrite the file I believe (e.g. in case someone specifies `--config somefile.txt` where somefile.txt is not a hubble config)

 Can we check for `viper.ConfigFileNotFoundError` as documented here: https://github.com/spf13/viper#reading-config-files